engine/src/execution_graph/logic_controllers/LogicalProject.cpp

```diff
@@ -132,6 +133,39 @@ std::unique_ptr<cudf::column> evaluate_string_functions(const cudf::table_view &
         computed_col = cudf::strings::replace(column, target, repl);
         break;
     }
+    case operator_type::BLZ_STR_REGEXP_REPLACE:
+    {
+        // required args: string column, pattern, replacement
+        // optional args: position, occurrence, match_type
+        assert(arg_tokens.size() >= 3 && arg_tokens.size() <= 6);
+        RAL_EXPECTS(arg_tokens.size() <= 4, "Optional parameters occurrence and match_type are not yet supported.");
+        RAL_EXPECTS(!is_literal(arg_tokens[0]), "REGEXP_REPLACE function not supported for string literals");
+
+        cudf::column_view column = table.column(get_index(arg_tokens[0]));
+        RAL_EXPECTS(is_type_string(column.type().id()), "REGEXP_REPLACE argument must be a column of type string");
+
+        std::string pattern = StringUtil::removeEncapsulation(arg_tokens[1], encapsulation_character);
+        std::string repl = StringUtil::removeEncapsulation(arg_tokens[2], encapsulation_character);
+
+        // handle the position argument, if it exists
+        if (arg_tokens.size() == 4) {
+            int32_t start = arg_tokens.size() == 4 ? std::stoi(arg_tokens[3]) : 0;
+            int32_t prefix = 0;
+
+            std::unique_ptr<cudf::column> prefix_col = cudf::strings::slice_strings(column, prefix, start);
+            cudf::column_view post_replace_view = cudf::strings::replace_with_backrefs(
+                cudf::column_view(cudf::strings::slice_strings(column, start)->view()),
+                pattern,
+                repl)->view();
+
+            computed_col = cudf::strings::concatenate(
+                cudf::table_view{{prefix_col->view(), post_replace_view}}
+            );
```

Comment:
The logic for this section is:
- If we have a `position` argument, we should only replace within the string starting from index `position`

To make this concrete, see the following MySQL example (note the indexing starts at 1). By default we replace all matches. With the position argument, we only replace valid matches starting from position 3 in this case, but still return the parts of the string before the we look for matches.

```
mysql> SELECT REGEXP_REPLACE('abc def ghi', '[a-z]+', 'X');
REGEXP_REPLACE('abc def ghi', '[a-z]+', 'X')
X X X
```
```mysql>SELECT REGEXP_REPLACE('abc def ghi', '[a-z]+', 'X', 3);
REGEXP_REPLACE('abc def ghi', '[a-z]+', 'X', 3)
abX X X
```

@davidwendt , do you have any thoughts on if there is a better way to do this than I have here?
- Slice the strings to get the characters before the `position` argument
- Slice the strings starting at the `position` argument and pass that to replace_with_backrefs
- Concatenate the results, recombining the "prefix" (portions of strings not sued for replace) with the "replaced" segment

Reply:
The logic looks right. I don't see a better way to do it. Maybe if we had a _split_on_position function_ you would not have to slice the strings twice (left and right).

The code has an error though. The `concatenate()` call may crash. You need to keep the `post_replace_view's` column alive until after `concatenate()` returns. Changing it to following would be safer:
```
        auto prefix_col = cudf::strings::slice_strings(column, prefix, start);
        auto post_replace_col = cudf::strings::replace_with_backrefs(
            cudf::column_view(cudf::strings::slice_strings(column, start)->view()),
            pattern,
            repl);

        computed_col = cudf::strings::concatenate(
            cudf::table_view{{prefix_col->view(), post_replace_col->view()}} );
```

Reply:
Thanks for confirming the logic, and for identifying and proposing a solution to the risky segment @davidwendt!

Reply:
Resolved by https://github.com/BlazingDB/blazingsql/pull/1200/commits/06ad76d9fbb11fd3f7b2746183c49c28b4cae612